src/System.Management.Automation/engine/runtime/Operations/NumericOps.cs

```diff
@@ -863,5 +863,34 @@ internal static object CompareIne(char lhs, char rhs)
             char secondAsUpper = char.ToUpperInvariant(rhs);
             return firstAsUpper != secondAsUpper ? Boxed.True : Boxed.False;
         }
+
+        internal static object[] Range(char start, char end)
+        {
+            int lower = (int)start;
+            int upper = (int)end;
+
+            if (lower == upper)
```

Comment:
I think this special case is correctly handled by the code below and it happens rarely, so maybe we should remove it. (And I see you copied the code from the `int` version, so you could remove it there too.)

Reply:
Mainly did it this way to keep it similar to the existing code in Compiler.cs. I'm inclined to make a single Range method that takes the operands as Expression's, de-duplicate the loop that generates the actual sequence and then cast the entire thing to char[], int[] or long[] based on the lhs operand before returning. Will update the code later today

Reply:
@IISResetMe Is this addresed?

Reply:
I've removed the redundant branch in both overloads now. https://github.com/PowerShell/PowerShell/pull/5026/commits/0406cb37f1556b8497325236672819b54da23e53